gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/resourcename/ResourceNameHelperClassComposer.java

```diff
@@ -160,8 +160,26 @@ private static List<AnnotationNode> createClassAnnotations() {
             .build());
   }
 
-  private static List<TypeNode> createImplementsTypes() {
-    return Arrays.asList(FIXED_TYPESTORE.get("ResourceName"));
+  /**
+   * Builds a list of types to be implemented. If the type has the same name then we use its full
+   * name
+   *
+   * @param implementingClassName class that is implementing the resulting list. Collisions are
+   *     checked
+   * @return
+   */
+  private static List<TypeNode> createImplementsTypes(String implementingClassName) {
+    List<TypeNode> preprocessed = Arrays.asList(FIXED_TYPESTORE.get("ResourceName"));
```

Comment:
`FIXED_TYPESTORE` always returns only one result. Can we check if there is conflict and fix it first, then wrap it as a list in the end?

Reply:
I used a slightly different but similarly concise way (wrap both possible results in the end). Hope you agree it's good too.

Reply:
Ah I only saw the review comment instead of this one. I'll leave it as a quick follow up